### PR TITLE
Adding barracks fort priority

### PR DESCRIPTION
**Pull request by dvincent56**
Barracks_UI_Mockup_copy_copy

### CODE REVIEW

**Comment by crudelios** [comment]
Instead of the cart image, you have enough space to add actual text next to the checkbox, which I think would be better. Or align the cart image/ checkbox left and add the text to the right of the checkbox.

**Comment by dvincent56** [comment]
Capture d'écran 2024-03-31 225725
Do you prefer the top right position like Grand Temple of Mars ?

**Comment by dvincent56** [comment]
Capture d'écran 2024-04-01 000048
Remove archery button and place delivery button on top right like grand temple

**Comment by crudelios** [comment]
This looks good but has merge conflicts now because of the campaign additions.

**Comment by dvincent56** [comment]
done

**Comment by dvincent56** [comment]
Areldir wants to revamp the toggle button.

**Comment by Areldir** [comment]
I'm a nuisance, I know lol

**Comment by dvincent56** [comment]
I found an issue, I pass the pull request to draft
@crudelios I put comments on my problem, do you know how can I fix it without copy past the military window code in culture window ?

**Comment by dvincent56** [comment]
@crudelios Do you have an idea to fix the grand temple update instead of last barracks ?

**Comment by dvincent56** [comment]
Find a way to fix but I don't know if it's a good way.

**Comment by dvincent56** [comment]
Breaks the save ...

File: src/building/building.h [hunk 1]
```diff
@@ -199,6 +199,7 @@ typedef struct building {
     unsigned char fumigation_direction;
     short resources[RESOURCE_MAX];
     unsigned char accepted_goods[RESOURCE_MAX];
+    unsigned char accept_delivery;
```
Comment (crudelios) [inline src/building/building.h:202]: Why not just use `accepted_goods[RESOURCE_WEAPONS]`?

That way you'd avoid the entire savegame hassle.

File: src/window/building_info.c [hunk 2]
```diff
@@ -453,7 +455,8 @@ static void init(int grid_offset)
         case 5: context.height_blocks = 24; break;
         case 6: context.height_blocks = 38; break;
         case 7: context.height_blocks = 26; break;
-        case 8: context.height_blocks = 40; context.width_blocks = 30; break;
+        case 8: context.height_blocks = 40; context.width_blocks = 30; break;        
+        case 10: context.height_blocks = 47; context.width_blocks = 30; break;
         case 9: context.height_blocks = 20; break;
```
Comment (crudelios) [inline src/window/building_info.c:460]: Extra tab at the end of the `case 8:` line

`case 9:` and `case 10:` lines swapped.


```suggestion
        case 8: context.height_blocks = 40; context.width_blocks = 30; break;
        case 9: context.height_blocks = 20; break;
        case 10: context.height_blocks = 47; context.width_blocks = 30; break;
```

File: src/window/building_info.c [hunk 3]
```diff
@@ -1062,6 +1065,8 @@ static void get_tooltip(tooltip_context *c)
         } else if (!context.depot_selection.resource) {
             window_building_depot_get_tooltip_main(&translation);
         }
+    } else if (btype == BUILDING_BARRACKS ||btype == BUILDING_GRAND_TEMPLE_MARS) {
```
Comment (crudelios) [inline src/window/building_info.c:1068]:
```suggestion
    } else if (btype == BUILDING_BARRACKS || btype == BUILDING_GRAND_TEMPLE_MARS) {
```

File: src/building/barracks.c [hunk 4]
```diff
@@ -79,6 +83,29 @@ static int get_closest_legion_needing_soldiers(const building *barracks)
     int recruit_type = LEGION_RECRUIT_NONE;
     int min_formation_id = 0;
     int min_distance = INFINITE;
+
+    int priority_recruitment = barracks->subtype.barracks_priority;
+    int requiered_recruitment = recruit_type;
```
Comment (crudelios) [inline src/building/barracks.c:88]:
```suggestion
    int required_recruitment = recruit_type;
```

Don't forget to change the variable name elsewhere.

**Review by crudelios — CHANGES_REQUESTED** [review]
Here are some needed changes.

Do note that there's an unfixed issue yet, on old savegames weapons will not be accepted by default.

I'm not sure how to fix that short of bumping the savegame version.

**Comment by crudelios** [comment]
Good job!

Now either a value of `0` means the good is accepted, which is unintuitive but would work without any change, or we set another control value besides `RESOURCE_WEAPONS` (`RESOURCE_NONE` comes to mind) to `1` to signal that the current savegame is already enabling/disabling weapons.

**Review by crudelios — APPROVED** [review]

**Comment by crudelios** [comment]
LGTM 👍

**Pull request merged**